packages/core/src/status-indicator/StatusIndicator.tsx

```diff
@@ -50,7 +44,21 @@ export const StatusIndicator = forwardRef<SVGSVGElement, StatusIndicatorProps>(
       window: targetWindow,
     });
 
-    const IconComponent = icons[status];
+    const {
+      SemanticErrorSolidIcon = ErrorSolidIcon,
```

Comment:
Shall we do these fallback logic within `useIcon` hook? So each components doesn't really need to know these specifics..?

If that's the case, we can also create linting rules to highlight, no `salt-ds/icon` import should be in core package...?

Reply:
added fallback logic in useIcon hook